x/tally/keeper/endblock.go

```diff
@@ -124,7 +127,13 @@ func (k Keeper) ProcessTallies(ctx sdk.Context, coreContract sdk.AccAddress) err
 			k.Logger(ctx).Debug("tally result", "request_id", req.ID, "tally_result", tallyResults[i])
 		}
 
-		processedReqs[req.ID] = distMsgs
+		baseFee := gasPrice.Mul(math.NewIntFromUint64(params.GasCostBase))
```

Comment:
I'll leave comments as I go through the PR rather than submitting them all at once.

I don't think this is correct. We should always start with the base fee so if the requestor did not attach enough gas the tally phase will run out of gas as expected during VM execution.